docs/designs/repository-urls.md

```diff
@@ -0,0 +1,208 @@
+# Repository Urls
+
+Explain the 3 types of repository urls used in DocFX and their design purposes and use scenarios.   
+
+*******
+
+## Definitions
+
+| Name | Definition |
+|------|------------|
+| Repository Url    | The url of current build repo, i.e. the argument of `git clone`  |
+| Repository Branch | The branch of current build repo    |
+| Publish Repository Url | The url of OPS provisioned repository |
+| Publish Repository Branch | The branch of OPS provisioned repository |
+| Pull Request Repository Url |  Present optionally, it will only present for builds triggered by pull requests |
+| Pull Request Repository Branch  | Present optionally, it will only present for builds triggered by pull requests  | 
+| Contribution Repository | Public repositories associated with the private provisioned repositories |
+
+
+	
+*******
+
+## Overview  
+The values of **Repository Url**, **Repository Branch**, **Publish Repository Url**, **Publish Repository Branch**, **Pull Request Repository Url** and **Pull Request Repository Branch**  will be passed via one or more the following approaches.
+1. DocFX command options
+2. environment variables
+3. config 
+
+For all DocFX build scenarios, it will always fetch build information via ****Publish Repository Url****.
```

Comment:
Currently, we have a fallback logic inside docfx, if the `publishedRepositoryUrl` doesn't exist, we will fallback to use `repositoryUrl`, I also want to mention that, only the `repositoryUrl` is required, the other two URL is optional.

Reply:
But if we put the fallback logic in the very beginning, after that, your word is still right.

Reply:
Let's call out the fallback logic as such design spec should focus more on DocFX input/output.